lib/screen/connection/persona_connections_page.dart

```diff
@@ -111,27 +120,98 @@ class _PersonaConnectionsPageState extends State<PersonaConnectionsPage>
     }
   }
 
+  void _showConnectionOption() async {
+    final options = [
+      OptionItem(
+          title: "connect_via_clipboard".tr(),
+          icon: SvgPicture.asset("assets/images/DApp.svg"),
+          onTap: () async {
+            if (!mounted) return;
+            Navigator.of(context).pop();
+            try {
+              final clipboardData = await Clipboard.getData("text/plain");
+              if (clipboardData == null ||
+                  clipboardData.text == null ||
+                  clipboardData.text!.isEmpty) {
+                throw ConnectionViaClipboardError("Clipboard is empty");
+              }
+              final text = clipboardData.text!;
+              log.info("Connect via clipboard: $text");
+              _processDeeplink(text);
+            } catch (e) {
+              if (e is ConnectionViaClipboardError) {
+                if (!mounted) return;
+                UIHelper.hideInfoDialog(context);
+                UIHelper.showInvalidURI(context);
+              } else {
+                rethrow;
+              }
+            }
+          }),
+      OptionItem(),
+    ];
+    UIHelper.showDrawerAction(context, options: options);
+  }
+
+  bool _isUriValid(String uri) {
+    if (uri.startsWith("wc:") || uri.length == TB_COPIED_IRL_LENGTH) {
```

Comment:
@ppupha The length might not be correct, for different marketplaces.
Can you use base58check decode? then parse json get the `id`?